src/pyhf/optimize/__init__.py

```diff
@@ -12,6 +12,13 @@ def __getattr__(self, name):
             # for autocomplete and dir() calls
             self.scipy_optimizer = scipy_optimizer
             return scipy_optimizer
+        elif name == 'autodiff':
```

Comment:
the `autodiff` is just a mixin not a standalone optimimzer so prob shouldn't appear here @kratsg

Reply:
it does need to be there for now.. until we update to python3 and fix the imports.

Reply:
#735 will address this.